src/main/scala/mesosphere/marathon/api/TaskKiller.scala

```diff
@@ -41,9 +42,10 @@ class TaskKiller @Inject() (
           val activeInstances = foundInstances.filter(_.isActive)
 
           if (wipe) {
-            await(Future.sequence(foundInstances.map(i => instanceTracker.setGoal(i.instanceId, Goal.Decommissioned)))): @silent
+            val instancesAreTerminal: Future[Done] = killService.watchForKilledInstances(activeInstances)
+            await(Future.sequence(foundInstances.map(i => instanceTracker.setGoal(i.instanceId, Goal.Decommissioned, GoalChangeReason.UserRequest)))): @silent
             await(expunge(foundInstances)): @silent
-            await(killService.killInstances(activeInstances, KillReason.KillingTasksViaApi)): @silent
+            await(instancesAreTerminal): @silent
```

Comment:
I think we still have the possibility of being stuck forever here; sending an expunge or a setGoal on something that is already terminal will not cause a new kill to be sent, so we won't issue a kill and we won't get a terminal task update.

Case should be rare; I think we can just put a timeout on watchForKilledInstances for now? We consider a TASK_UNREACHABLE status to be "terminal" enough here, from what I remember.